### ISSUE TITLE
SITL Cygwin Install  "C:\cygwin\home" folder empty

### ISSUE THREAD

**Issue by jhenry5** [ArduPilot/ardupilot_wiki#521]
Hi, I followed the instructions http://ardupilot.org/dev/docs/sitl-native-on-windows.html  to install 32 bit cygwin on my 64 bit windows 8.1 laptop. My next step is to open my .bashrc file located in C:\cygwin\home\ & add a line to it. The problem I'm having is that " C:\cygwin\home\" is empty when i open it in windows explorer.
1) I unhid all hidden files. 
2) I re-installed cygwin for a total of three installs from different servers.
I cant find .bashrc file to modify, its not in C:\cygwin\home. **This folder is empty.**Please help

![image](https://cloud.githubusercontent.com/assets/5570690/17874674/3d6df050-689b-11e6-829c-9764dc3e631e.png)

![image](https://cloud.githubusercontent.com/assets/5570690/17874701/5e882a94-689b-11e6-8ecc-14e97de6cdb3.png)

**Comment by hamishwillee** [comment]
Hi @jhenry5 

Firstly, best place to get support for development is the [discussion boards](http://discuss.ardupilot.org/). That said, this is probably a docs issue too.

So the file should actually be C:/cygwin/home/**user_name**/.bashrc , but obviously that doesn't help you since your home folder is empty. Perhaps the best thing to do is work out where cygwin things your home folder is. Please open cygwin terminal and type the following:
```
cd ~
pwd
ls -la
```
The first line takes you to your home directory, the second line tells you where that is and the third lists all files. If it includes a .bashrc file then you know it exists and can search for the folder name in your normal drive (or edit via the prompt).

If the .bashrc is not there, then we have a problem. Let me know what you see.

**Comment by jhenry5** [comment]
Thanks kindly,
The home folder isnt where i expect it to be. But the files, i require arent in this folder either.

![image](https://cloud.githubusercontent.com/assets/5570690/17876530/fe43768a-68a8-11e6-959f-9a9f12e1758b.png)

Looking at the file creation dates bash.history is the only file modified today. (8/22/2016)

Any feedback will be greatly appreciated

**Comment by hamishwillee** [comment]
Hi @jhenry5 

That is very odd - it might be worth uninstalling/reinstalling Cygwin. 

When you first opened the cygwin terminal (step 1 in "Set up directories/paths in Cygwin") it should have showed you where it was creating the .bashrc file. If it didn't this step may have failed. I'd be searching for the file across my C drive.

Other than that I see two options:

1. Just create a file .bashrc in that directory and add the information. This may work, but I'm concerned that there may be other useful information in the default .bashrc that you are missing.
2. Start using Google search to find out how to change the home folder.  E.g. [see this](http://stackoverflow.com/questions/1494658/how-can-i-change-my-cygwin-home-folder-after-installation). You would probably do this anyway since a dia folder is not an ideal home location.

I think that's pretty much the help I can give you - Cygwin is just part of the toolchain - not really an "ardupilot" problem.

**Comment by jhenry5** [comment]
cygwin doesnt have an uninstall option and does not show up in my add remove programs.
I deleted the "C/Cygwin" folder & tried again, without success.....My install selections are below:
![image](https://cloud.githubusercontent.com/assets/5570690/17877075/1ec4565e-68ae-11e6-9054-2b7c1aedf954.png)
![image](https://cloud.githubusercontent.com/assets/5570690/17877079/27d02462-68ae-11e6-8d6f-0af59860af01.png)

![image](https://cloud.githubusercontent.com/assets/5570690/17877104/5998a1a4-68ae-11e6-8731-7a81dd7fb366.png)

![image](https://cloud.githubusercontent.com/assets/5570690/17877304/3198ebbc-68b0-11e6-8c3c-bb32e2f9970b.png)

**The full print out of my packages**
autoconf2.1	(2.13-12)
	Stable version of the automatic configure script builder
	Required by: autoconf, automake1.4, automake1.5

autoconf2.5	(2.69-3)
	Current version of the automatic configure script builder
	Required by: autoconf, automake1.10, automake1.11, automake1.12, automake1.13, automake1.14, automake1.15, automake1.6, automake1.7, automake1.8, automake1.9

automake1.10	(1.10.3-2)
	(1.10) a tool for generating GNU-compliant Makefiles
	Required by: automake

automake1.11	(1.11.6-2)
	(1.11) a tool for generating GNU-compliant Makefiles
	Required by: automake

automake1.12	(1.12.6-2)
	(1.12) a tool for generating GNU-compliant Makefiles
	Required by: automake

automake1.13	(1.13.4-1)
	(1.13) a tool for generating GNU-compliant Makefiles
	Required by: automake

automake1.14	(1.14.1-2)
	(1.14) a tool for generating GNU-compliant Makefiles
	Required by: automake

automake1.15	(1.15-1)
	(1.15) a tool for generating GNU-compliant Makefiles
	Required by: automake

automake1.4	(1.4p6-11)
	(1.4) a tool for generating GNU-compliant Makefiles
	Required by: automake

automake1.5	(1.5-11)
	(1.5) a tool for generating GNU-compliant Makefiles
	Required by: automake

automake1.6	(1.6.3-12)
	(1.6) a tool for generating GNU-compliant Makefiles
	Required by: automake

automake1.7	(1.7.9-11)
	(1.7) a tool for generating GNU-compliant Makefiles
	Required by: automake

automake1.8	(1.8.5-11)
	(1.8) a tool for generating GNU-compliant Makefiles
	Required by: automake

automake1.9	(1.9.6-11)
	(1.9) a tool for generating GNU-compliant Makefiles
	Required by: automake

binutils	(2.25-4)
	GNU assembler, linker, and similar utilities
	Required by: gcc-core, python

bzip2	(1.0.6-2)
	A high-quality block-sorting file compressor (utilities)
	Required by: tar

ca-certificates	(2.8-1)
	CA root certificates
	Required by: libopenssl100, libcurl4

csih	(0.9.9-1)
	Provides support for installing cygwin services
	Required by: openssh

cygrunsrv	(1.62-1)
	NT/W2K service initiator
	Required by: openssh

cygwin-devel	(2.5.2-1)
	Core development files
	Required by: gcc-core, python

diffutils	(3.3-3)
	A GNU collection of diff utilities
	Required by: openssh, csih

gcc-core	(5.4.0-1)
	GNU Compiler Collection (C, OpenMP)
	Required by: gcc-g++

groff	(1.22.3-1)
	GNU roff formatter
	Required by: man-db

less	(481-1)
	A file pager program, similar to more(1)
	Required by: git, man-db

libargp	(20110921-2)
	Interface for parsing command line arguments
	Required by: getent

libatomic1	(5.4.0-1)
	GCC C11/C++11 locked atomics runtime library
	Required by: gcc-core

libattr1	(2.4.46-1)
	Shared lib for managing filesystem extended attributes
	Required by: coreutils

libblkid1	(2.25.2-2)
	Block device ID library (runtime)
	Required by: util-linux

libbz2_1	(1.0.6-2)
	A high-quality block-sorting file compressor (runtime)
	Required by: python, bzip2

libcharset1	(1.14-3)
	Unicode iconv() implementation
	Required by: libiconv-devel

libcom_err2	(1.42.12-2)
	Common error description library (runtime)
	Required by: libgssapi_krb5_2, libkrb5_3

libcrypt0	(1.4-1)
	Encryption/Decryption utility and library
	Required by: openssh, perl_base, python, libguile17, perl, libsasl2_3

libcurl4	(7.50.1-1)
	Multi-protocol file transfer library (runtime)
	Required by: git

libdb4.8	(4.8.30-1)
	Oracle Berkeley DB (db4.8 - C and C++ libs)
	Required by: python, perl, libsasl2_3

libedit0	(20130712-1)
	The NetBSD Editline library (runtime)
	Required by: openssh

libexpat1	(2.2.0-0)
	Expat XML parser library (shared library)
	Required by: git, libexpat-devel, python

libffi6	(3.2.1-1)
	Portable foreign function interface library
	Required by: python, libp11-kit0

libgcc1	(5.4.0-1)
	GCC C runtime library
	Required by: bash, ccache, coreutils, file, findutils, gawk, git, grep, gzip, libopenssl100, man-db, tar, util-linux, vim-minimal, libncursesw10, zlib0, libpopt0, libmpfr4, gcc-core, libstdc++6, less, libcurl4, openssh, perl_base, python, libguile17, groff, bzip2, xz, libblkid1, libsmartcols1, libuuid1, m4, perl, libgomp1, libquadmath0, diffutils, libkrb5_3, libbz2_1, libdb4.8, perl-Unicode-Normalize, libcom_err2, libunistring2, libtasn1_6

libgdbm4	(1.8.3-20)
	GNU dbm database routines (runtime)
	Required by: man-db, python, perl

libgmp10	(6.1.0-3p1)
	Library for arbitrary precision arithmetic (C runtime)
	Required by: coreutils, gawk, gcc-g++, libmpfr4, gcc-core, libisl13, libmpc3, libguile17

libgomp1	(5.4.0-1)
	GCC OpenMP runtime library
	Required by: gcc-core

libgssapi_krb5_2	(1.14.1-2)
	Kerberos reference implementation GSS-API library
	Required by: libcurl4, openssh, libsasl2_3

libguile17	(1.8.8-1)
	GNU Scheme interpreter library
	Required by: make

libiconv	(1.14-3)
	Unicode iconv() implementation
	Required by: man-db

libiconv-devel	(1.14-3)
	Unicode iconv() implementation
	Required by: libintl-devel

libiconv2	(1.14-3)
	Unicode iconv() implementation
	Required by: alternatives, bash, coreutils, gcc-g++, git, grep, info, man-db, tar, vim-minimal, libintl8, libpopt0, gcc-core, rsync, groff, libiconv, libidn11, libpsl5, diffutils, libunistring2, libiconv-devel

libidn11	(1.29-1)
	International Domain Name library (runtime)
	Required by: libcurl4, libpsl5

libintl-devel	(0.19.7-1)
	GNU Internationalization runtime library
	Required by: libuuid-devel

libintl8	(0.19.7-1)
	GNU Internationalization runtime library
	Required by: alternatives, bash, coreutils, findutils, gawk, gcc-g++, git, grep, info, make, man-db, sed, tar, util-linux, libpopt0, gcc-core, python, libguile17, libiconv, xz, libblkid1, libsmartcols1, libuuid1, binutils, libgssapi_krb5_2, libidn11, diffutils, libkrb5_3, libk5crypto3, libkrb5support0, libintl-devel, libp11-kit0

libisl13	(0.14.1-1)
	Integer Set Library (runtime)
	Required by: gcc-g++, gcc-core

libk5crypto3	(1.14.1-2)
	Kerberos reference implementation crypto library
	Required by: libgssapi_krb5_2, libkrb5_3

libkrb5support0	(1.14.1-2)
	Kerberos reference implementation support library
	Required by: libgssapi_krb5_2, libkrb5_3, libk5crypto3

libkrb5_3	(1.14.1-2)
	Kerberos reference implementation library
	Required by: openssh, libgssapi_krb5_2, libsasl2_3

libltdl7	(2.4.6-4)
	Libtool dynamic library loader library (runtime)
	Required by: libtool, libguile17

liblzma5	(5.2.2-1)
	LZMA de/compressor library (runtime)
	Required by: xz

libmpc3	(1.0.3-1)
	C library for multiple-precision floating-point computations with exact rounding (runtime)
	Required by: gcc-g++, gcc-core

libmpfr4	(3.1.4-1)
	A library for multiple-precision floating-point arithmetic with exact rounding (runtime)
	Required by: gawk, gcc-g++, gcc-core, libmpc3

libncursesw10	(6.0-7.20160806)
	Terminal display library
	Required by: bash, info, libreadline7, ncurses, util-linux, vim-minimal, less, python, libedit0

libnghttp2_14	(1.7.1-1)
	HTTP/2 C library
	Required by: libcurl4

libopenldap2_4_2	(2.4.42-1)
	Lightweight Directory Access Protocol suite (runtime)
	Required by: libcurl4, libsasl2_3

libp11-kit0	(0.22.1-1)
	PKCS#11 module library
	Required by: p11-kit

libpcre1	(8.39-1)
	Perl Compatible Regular Expressions UTF-8 runtime
	Required by: git, grep

libpipeline1	(1.4.0-1)
	C library for manipulating pipelines of subprocesses - runtime
	Required by: man-db

libpopt-common	(1.16-2)
	Command-line option parser library
	Required by: libpopt0

libpopt0	(1.16-2)
	Command-line option parser library
	Required by: cygutils

libpsl5	(0.13.0-2)
	Public Suffix List library
	Required by: libcurl4

libquadmath0	(5.4.0-1)
	GCC Quad-Precision Math runtime library
	Required by: gcc-core

libsasl2_3	(2.1.26-9)
	The Cyrus SASL API implementation. (Runtime library and Daemon)
	Required by: libopenldap2_4_2

libsigsegv2	(2.10-2)
	Library for handling page faults in user mode (runtime)
	Required by: m4, diffutils

libsmartcols1	(2.25.2-2)
	Tabular data formatting library (runtime)
	Required by: util-linux

libsqlite3_0	(3.14.1-1)
	An embeddable SQL database engine (library)
	Required by: python

libssh2_1	(1.7.0-1)
	SSH2 protocol library
	Required by: libcurl4

libssp0	(5.4.0-1)
	GCC Stack-Smashing Protection runtime library
	Required by: vim-minimal, gcc-core, openssh, perl-TermReadKey, perl_base, perl, perl-Unicode-Normalize

libstdc++6	(5.4.0-1)
	GCC C++ runtime library
	Required by: gcc-g++, libncursesw10, groff, libdb4.8

libtasn1_6	(4.9-1)
	ASN.1 library (runtime)
	Required by: p11-kit, p11-kit-trust

libunistring2	(0.9.6-1)
	Library for manipulating Unicode strings - runtime
	Required by: libpsl5

libuuid-devel	(2.25.2-2)
	Universally Unique ID library (development)
	Required by: python

libuuid1	(2.25.2-2)
	Universally Unique ID library (runtime)
	Required by: util-linux, libblkid1, libuuid-devel

libvtv0	(5.4.0-1)
	GCC vtable verification library
	Required by: gcc-core

m4	(1.4.17-2)
	GNU implementation of the traditional Unix macro processor
	Required by: autoconf2.1, autoconf2.5, automake1.10, automake1.11, automake1.12, automake1.13, automake1.14, automake1.15, automake1.4, automake1.5, automake1.6, automake1.7, automake1.8, automake1.9

openssh	(7.3p1-1)
	The OpenSSH server and client programs
	Required by: git

p11-kit	(0.22.1-1)
	PKCS#11 module tool
	Required by: ca-certificates

p11-kit-trust	(0.22.1-1)
	PKCS#11 module library
	Required by: ca-certificates

perl	(5.22.2-1)
	Perl programming language interpreter
	Required by: autoconf2.1, autoconf2.5, automake1.10, automake1.11, automake1.12, automake1.13, automake1.14, automake1.4, automake1.5, automake1.6, automake1.7, automake1.8, automake1.9, texinfo, perl-Pod-Simple, perl-Unicode-Normalize

perl-Carp	(1.38-1)
	Perl distribution Carp
	Required by: git, automake1.14, automake1.15

perl-Error	(0.17024-1)
	Perl distribution Error
	Required by: git

perl-Pod-Simple	(3.32-1)
	Perl distribution Pod-Simple
	Required by: texinfo

perl-TermReadKey	(2.33-1)
	Perl distribution TermReadKey
	Required by: git

perl-Unicode-Normalize	(1.25-1)
	Perl distribution Unicode-Normalize
	Required by: texinfo

perl_autorebase	(5.22.2-1)
	Perl programming language interpreter
	Required by: perl

perl_base	(5.22.2-1)
	Perl programming language interpreter
	Required by: git, automake1.14, automake1.15, perl-Carp, perl-Error, perl-TermReadKey, perl, texinfo, perl-Pod-Simple, perl-Unicode-Normalize

python	(2.7.10-1)
	Python language interpreter
	Required by: git

rsync	(3.1.2-1)
	Fast remote file transfer program (can use existing data to minimize transfer)
	Required by: git

texinfo	(6.1-2)
	GNU documentation formatter
	Required by: automake1.10, automake1.11, automake1.12, automake1.13, automake1.14, automake1.15, automake1.4, automake1.5, automake1.6, automake1.7, automake1.8, automake1.9

w32api-headers	(4.0.4-1)
	MinGW-w64 Windows API headers for Cygwin
	Required by: gcc-core, w32api-runtime

w32api-runtime	(4.0.4-1)
	MinGW-w64 Windows API import libraries for Cygwin
	Required by: gcc-core

windows-default-manifest	(6.4-1)
	Default Windows application manifest
	Required by: gcc-core

xz	(5.2.2-1)
	LZMA de/compressor
	Required by: tar, automake1.10, automake1.11, automake1.12, automake1.13, automake1.14, automake1.15

zlib-devel	(1.2.8-3)
	gzip de/compression library (development)
	Required by: binutils

zlib0	(1.2.8-3)
	gzip de/compression library (runtime)
	Required by: ccache, file, gcc-g++, git, libopenssl100, man-db, util-linux, gcc-core, libcurl4, openssh, python, binutils, libssh2_1, zlib-devel

**Comment by hamishwillee** [comment]
Try installing it for "just you". I'll bet that might set the default to your home directory.

**Comment by jhenry5** [comment]
I reinstalled cygwin 4 times and C:\cygwin\home is still empty.  I then tried it again with the "just for you" setting, unsuccessfully. How do I generate all the files(including .bashrc) for C:\cygwin\home folder. 1)Can you point me(& anyone else with this problem) in the right direction to generate this file?﻿ 2) Where can you find the cigwyn uninstall "button" ?

Thx alot mate, I've been at this from 8 am to midnight

**Comment by hamishwillee** [comment]
Hi @jhenry5 

Sorry you're having so much trouble. Google is your friend!!!

I found [this post](http://superuser.com/questions/584540/no-bashrc-file-in-my-home-directory) which suggests that the default files are found in `\etc\skel`. Doing a search I find them in my drive at `C:\cygwin\etc\skel`. I also find that there is no "button" for uninstalling Cygwin - but there are some instructions here: https://cygwin.com/faq/faq.html#faq.setup.uninstall-all

So you first need to set your home and then move these files in there.

Good luck

Regards
H

**Comment by peterbarker** [comment]
@jhenry5 Were your problems eventually resolved?

Are there any modifications we need to make the the ArduPilot Wiki so this doesn't happen to someone else?

**Comment by kshitijvyas1985** [comment]
Any body facing this issue. Just set there environment variable
Right click on "My Computer" -> Properties -> Advanced -> Environment Variables
Add a new environment variable, called CYGWIN_HOME and set its value to C:\cygwin
Edit the PATH environment variable and add C:\cygwin\bin to it (usually separated by a ';').
Just click okay, exit any command prompts or bash shells (over cygwin) you may have open, and open it again - it'll work!
Assumption - this assumes that you have installed cygwin at C:\cygwin. If you've kept it someplace else, please modify the above accordingly

**Comment by brunoolivieri** [comment]
@jhenry5 Were your problems eventually resolved?

@kshitijvyas1985 I did the PR #1930 in order to emphasize the path configuration.

Moreover, this page has been updated since the issue opening. Makes sense remain with it open?

**Comment by Hwurzburg** [comment]
this seems resolved...closing

**Comment by ieliopou0** [comment]
Just adding this comment in case anyone else has this issue.
I also had this issue, where  C:\cygwin64\home was empty. The terminal displayed "-bash-4.4$" as the prompt, and some basic commands wouldn't work like "ls" and "clear". Adding the cygwin64 bin folder to the PATH environment variable like kshitijvyas1985 mentioned helped get some commands like "ls" to work, but home folder was till empty, "clear" command still didn't work, and the terminal prompt was still weird. Looked up how to at least get the "clear" command to work, and someone mentioned install the "ncurses" packages. So I re-ran the installer, kept the same folders I used to install cygwin originally, and then looked for packages that were not installed and searched for "ncurses", then installed the latest version of all the packages listed. I also noticed just before looking for ncurses stuff that there were a lot of packages listed under "pending". Either way I hit install, and this time when I opened up my cygwin terminal it looked normal, and the home directory had the .bashrc and other expected files under home/<user>. I'm new to cygwin so I don't know if those pending packages meant they weren't installed yet, and I had to re-run the installer to install them, or if it was due to the ncurses stuff, but it worked for me. Hopefully it works for anyone else with this issue.
